### PR TITLE
[NCL-6993] Add REST endpoint for target repositories

### PR DESCRIPTION
**Pull request by pkocandr**
### Checklist:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/pnc/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

File: rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/TargetRepositoryEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+                            content = @Content(schema = @Schema(implementation = ErrorResponse.class))) })
+    @GET
+    @Path("/{id}")
+    @Consumes(MediaType.APPLICATION_JSON_PATCH_JSON) // workaround for PATCH support
+    TargetRepository getSpecific(@Parameter(description = REPO_ID) @PathParam("id") String id);
+
+    static final String CREATE_NEW_DESC = "Creates a new target repository.";
+
+    /**
+     * {@value CREATE_NEW_DESC} {@value CREATE_NEW_DESC2}
```
Comment (janinko) [inline rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/TargetRepositoryEndpoint.java:134]: there is no CREATE_NEW_DESC2

File: rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/TargetRepositoryEndpoint.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+                            responseCode = CONFLICTED_CODE,
+                            description = CONFLICTED_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = ErrorResponse.class))),
+                    @ApiResponse(
+                            responseCode = SERVER_ERROR_CODE,
+                            description = SERVER_ERROR_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = ErrorResponse.class))) })
+    @POST
+    @RespondWithStatus(Response.Status.CREATED)
+    TargetRepository createNew(@NotNull TargetRepository request);
```
Comment (janinko) [inline rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/TargetRepositoryEndpoint.java:160]: Shouldn't this endpoint be limited to admins?

**Review by janinko — CHANGES_REQUESTED** [review]

File: rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/TargetRepositoryEndpoint.java (hunk 2 above)
Comment (pkocandr) [reply rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/TargetRepositoryEndpoint.java:160]: Yes, it definitely should. I forgot to limit it. Thank you!

File: rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/TargetRepositoryEndpoint.java (hunk 1 above)
Comment (pkocandr) [reply rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/TargetRepositoryEndpoint.java:134]: OK, I have no idea where it came from... :-) Copy-paste error probably.

File: rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/TargetRepositoryEndpoint.java [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+     * @param request
+     * @return
+     */
+    @Operation(
+            summary = "[role:admin] " + CREATE_NEW_DESC,
+            responses = {
+                    @ApiResponse(
+                            responseCode = ENTITY_CREATED_CODE,
+                            description = ENTITY_CREATED_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = RepositoryCreationResponse.class))),
```
Comment (janinko) [inline rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/TargetRepositoryEndpoint.java:145]: Its's not repository

**Review by janinko — CHANGES_REQUESTED** [review]

File: rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/TargetRepositoryEndpoint.java (hunk 3 above)
Comment (pkocandr) [reply rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/TargetRepositoryEndpoint.java]: Fixed along with javadocs referencing SCM_ID.

**Comment by pkocandr** [comment]
retest this please

**Pull request merged**